docs/source/concepts/interoperability.rst

```diff
@@ -0,0 +1,237 @@
+
+To keep the transaction stream consistent, `P2` additionally outputs a ``LeaveLocalLedgerEvent`` on Alice's transaction stream.
+This event signals that the endpoint no longer outputs events concerning this contract; in particular not when the contract is archived.
+The contract is no longer reported in the active contract service and cannot be used by command submissions.
+
+Conversely, `P1` outputs an ``EnterLocalLedgerEvent``\ s some time before the ``ArchivedEvent`` on the transaction stream.
+This event signals that the endpoint starts outputting events concerning this contract.
+The contract is reported in the active contract service and can be used by command submission.
+The ``EnterLocalLedgerEvent`` contains all the information in a ``CreatedEvent``;
+the only difference is that ``EnterLocalLedgerEvent``\ s may occur several times whereas there should be at most one ``CreatedEvent`` for each contract.
+
+These events are generated when the underlying interoperability protocol synchronizes between the different ledgers.
+This may happen as part of command submission or for other reasons, e.g., load balancing.
+It is guaranteed that the ``EnterLocalLedgerEvent`` precedes contract usage, subject to the trust assumptions of the underlying ledgers and the interoperability protocol.
+
+A contract may enter and leave the local ledger of a Participant Node several times.
+For example, suppose that Bob submits the following commands and their commits end up on the given ledgers.
+
+#. Create a contract `c` with signatories Alice and Bob on Ledger 2
+#. Exercise a non-consuming choice `ch1` on `c` on Ledger 1.
+#. Exercise a non-consuming choice `ch2` on `c` on Ledger 2.
+#. Exercise a consuming choice on `c` on Ledger 1.
+
+Then, the transaction tree stream that `P2` provides for `A` contains five events involving contract `c`: ``EnterLocalLedgerEvent``, ``ExercisedEvent``, ``LeaveLocalLedgerEvent``, ``EnterLocalLedgerEvent``, ``ArchivedEvent``.
+These five events can be grouped into between two or five transactions.
+For example, the first three and the last two could be batched into one transaction each.
+However, `P2` cannot combine the ``LeaveLocalLedgerEvent`` with the subsequent ``EnterLocalLedgerEvent`` and `P2` must not elide them either.
+This is because their presence indicates that `P2`\ 's local ledger for Alice may miss some events in between; in this example, exercising the choice `ch2`.
+
+The flat transaction stream by `P2` shows omits the non-consuming exercise choices.
+It nevertheless contains the three view change events ``EnterLocalLedgerEvent``, ``LeaveLocalLedgerEvent``, and ``EnterLocalLedgerEvent`` before the ``ArchivedEvent``.
+This is because the endpoint cannot know at the ``LeaveLocalLedgerEvent`` that there will be another ``EnterLocalLedgerEvent``.
+
+In contrast, `P1` need not output the ``EnterLocalLedgerEvent``\ s and ``LeaveLocalLedgerEvent``\ s at all in this example because `P1` hosts `A` on both ledgers.
```

Comment:
I'm not so hot on this cancellation of enter and leave. I think we need a way to expose the current domain to the application, in case it needs to handle transfers etc manually. 

However, I see that the cancellation it might provide a more gentle migration path for apps, if they are used in deployments where a party is hosted only on a single participant. So maybe we want two channels, one that cancels, and one that doesn't.